js/src/interface.js

```diff
@@ -85,22 +85,24 @@ let app = new Vue({
       });
     }
 
-    Fliplet.Studio.onMessage((event) => {
-      if (event.data && event.data.event === 'overlay-close') {
-        this.reloadDataSources(event.data.data.dataSourceId);
-      }
-    });
-
     this.getDataSources().then(() => {
       this.isLoading = false;
     }).catch(() => {
       this.isLoading = false;
     });
+
     Fliplet.Widget.autosize();
   },
+  updated() {
+    Vue.nextTick(() => {
+      if (!this.dataSourceProvider) {
+        const dataSourceID = initialResult && initialResult.dataSourceId;
+        this.initDataSourceProvider(dataSourceID);
```

Comment:
@upplabs-alex-levchenko Can you confirm you have reviewed the latest changes?

```suggestion

        this.initDataSourceProvider(dataSourceID);
```

Reply:
@tonytlwu Yes I reviewed. I didn't check formatting since it is enough auto-formatting tools, please configure them and let's not waste time on this anymore. Thanks.

Reply:
@upplabs-alex-levchenko The code has been fixed by @AndrRyaz so it's no longer an issue.

Reply:
Roger that